### DISCUSSION TITLE
About gnome software over fwupd in snapd

### DISCUSSION THREAD

**Discussion by vincent-tang1** [fwupd/fwupd#6686] [General]
According fwupd README description,
```
Users should just have fwupd installed and updated by their distro, managed and tested by the package maintainer. The distribution will have also done some testing with how fwupd interacts with other software on your system, for instance using GNOME Software.

Installing fwupd using [Snap](https://github.com/fwupd/fwupd/wiki/fwupd-snap) or using [Flatpak](https://github.com/fwupd/fwupd/wiki/fwupd-flatpak) might be useful to update a specific device on the command line that needs a bleeding edge fwupd version, but it should not be considered as a replacement to the distro-provided system version.
```

not sure if it is your expectation that fwupd installed by snap can be used for GUI GNOME Software ? thanks

**Comment by hughsie** [comment]
It is the expectation, but personally I've been underwhelmed by the snap. I'm always lobbying distros to update the packaged fwupd version.

**Reply by vincent-tang1** [reply]
so, gnome software can automaticlly run below client fwupd utility (by snap install result), my understanding is correct?
```
fwupd.fwupdmgr
``` 

and if fwupd installed snap is used, it always show below warning, right ?
```
root@FWA-6172-efi:~# fwupd.fwupdmgr get-devices
WARNING: This package has not been validated, it may not work properly.
Advantech FWA-6172
│
├─BIOS:
...
```

**Reply by superm1** [reply]
> I'm always lobbying distros to update the packaged fwupd version

The thing I really like about snap is that besides confinement issues we fully control it upstream.  I think once we can escape any confinement issues it's a good alternative.

> personally I've been underwhelmed by the snap

It's getting a lot closer to usable now.  In the future we can advocate using this to transition to replace distro package.
https://github.com/fwupd/fwupd-snap-deb/

**Comment by superm1** [comment]
There is an issue where it doesn't auto connect the dbus interface.

https://github.com/fwupd/fwupd/issues/6678

Once the auto connect is done I expect it works.

**Comment by vincent-tang1** [comment]
> There is an issue where it doesn't auto connect the dbus interface.

thanks, so not only gnome firmware APP, but also gnome software APP  also have same stiuation, right ?

**Reply by superm1** [reply]
That's correct.  Once those auto-connect actions are done from the store I expect it goes away.  You can manually connect for now.

**Comment by vincent-tang1** [comment]
```
root@FWA-6172-efi:~# fwupd.fwupdmgr get-devices
WARNING: This package has not been validated, it may not work properly.
Advantech FWA-6172
...
```
not sure if this warning is your expectation and always exist in fwupd installed snap whatever any versions (edge/candiate/stable) ?

**Comment by hughsie** [comment]
> WARNING: This package has not been validated, it may not work properly.

We expect that from anything that isn't built from a tag, or supported branch.
